### PR TITLE
Create Quay teams that don't already exist

### PR DESCRIPTION
**Pull request by steveteahan**
This change creates a Quay team when it doesn't exist and a member needs to be added to the team.  A default role of `member` is assigned to the team as a sane default.  Prior to this change an exception would be thrown and the team would need to be created manually.

### CODE REVIEW

**Comment by steveteahan** [comment]
Additional app-sre specific testing is documented in: https://issues.redhat.com/browse/APPSRE-3488

File: reconcile/quay_membership.py [hunk 1]
```diff
@@ -67,12 +67,19 @@ def fetch_current_state(quay_api_store):
...
-            state.add({
-                'service': 'quay-membership',
-                'org': org_key,
-                'team': team
-            }, members)
+            try:
+                members = quay_api.list_team_members(team)
+            except QuayTeamNotFoundException:
+                logging.warning("Attempted to list members for team %s/%s, "
+                                "but it doesn't exist", org_key.org_name, team)
```
Comment (BumbleFeng) [inline reconcile/quay_membership.py:74]: Could you include org_key.instance in the log?

File: reconcile/utils/quay_api.py [hunk 2]
```diff
@@ -90,6 +101,33 @@ def add_user_to_team(self, user, team):
             raise RequestsException(r)
         return True
 
+    def create_or_update_team(self, team: str, role="member",
+                              description=None) -> None:
```
Comment (BumbleFeng) [inline reconcile/utils/quay_api.py:105]: Could you keep the same format with other methods and remove the team type and return type? I just want or keep consistency.

**Review by BumbleFeng — APPROVED** [review]
Generally LGTM, good job

File: reconcile/utils/quay_api.py (hunk 2 above)
Comment (steveteahan) [reply reconcile/utils/quay_api.py:105]: You'd like me to remove the docstring and type hints?  My reasons for adding those were:

- The docstring provides a link to the Quay API docs which might be useful for quick references?  It also documents which exceptions can be raised.
- Gradually introducing the type hints can assist IDEs in auto-completion and prepare the code base for the opportunity to use tools like mypy, pyre, etc. in the future.

I understand this affects the consistency of the code, but it opens up the opportunity to improve the documentation of the other methods next time a change needs to be made to them.

File: reconcile/quay_membership.py (hunk 1 above)
Comment (steveteahan) [reply reconcile/quay_membership.py:74]: Good call, I'll fix that!

**Comment by steveteahan** [comment]
Thank you for the review @BumbleFeng!

File: reconcile/utils/quay_api.py (hunk 2 above)
Comment (BumbleFeng) [reply reconcile/utils/quay_api.py:105]: works for me

**Pull request merged**